src/query_plan/operation.rs

```diff
@@ -2985,6 +3077,31 @@ impl NormalizedField {
         // @defer cannot be on field at the moment
         false
     }
+
+    pub(crate) fn type_if_added_to(
+        &self,
+        parent_type: &CompositeTypeDefinitionPosition,
+        schema: &ValidFederationSchema,
+    ) -> Option<CompositeTypeDefinitionPosition> {
+        let data = self.data();
+        if &data.field_position.parent() == parent_type && &data.schema == schema {
```

Comment:
stylistic nit but writing the references the other way around makes this a bit easier to read IMO, so you avoid the `&& &`. It functions the same, since `==` operates on a reference it doesn't actually need to move out of the reference when you write `*parent_type`.
```suggestion
        if data.field_position.parent() == *parent_type && data.schema == *schema {
```